src/Components/Shifting/BadgesList.tsx

```diff
@@ -1,80 +1,41 @@
-import { getAnyFacility, getUserList } from "../../Redux/actions";
-import { useEffect, useState } from "react";
-
 import { SHIFTING_FILTER_ORDER } from "../../Common/constants";
-import { useDispatch } from "react-redux";
+
 import { useTranslation } from "react-i18next";
+import useQuery from "../../Utils/request/useQuery";
+import routes from "../../Redux/api";
+
+function useFacilityQuery(facilityId: string | undefined) {
+  return useQuery(routes.getAnyFacility, {
+    pathParams: { id: String(facilityId) },
+    prefetch: facilityId !== undefined,
+  });
+}
```

Comment:
Why stringify when you are already passing a string? Consider casting it like this. If you do `String(undefiend)` you'll get `"undefined"`

```suggestion
function useFacilityQuery(facilityId?: string) {
  return useQuery(routes.getAnyFacility, {
    pathParams: { id: facilityId as string },
    prefetch: !!facilityId,
  });
}
```